### ISSUE TITLE
dot syntax vs square brackets

### ISSUE THREAD

**Issue by test-fullautomation** [test-fullautomation/python-jsonpreprocessor#86]
Assumed componentB_param1 has an array containing a data structure 

```
allowed
${componentB_param_1}[0]  
        ${componentB_param_1.0.abc.def} 

        nested dot syntax

        ${componentB_param_1.${param_a.0.B}.abc.def}

        nested square bracket syntax

        ${componentB_param_1}[${param_a}[0]["b"]]["abc"]["def"]
```              

Further important rules:
-	No syntax mix within an expression in one line (either dots only or square brackets only)
-	No dots in key names within dot syntax use (in this case the square brackets syntax must work)

-	`${componentB_param_1.0.abc[0]}` not allowed
        `${componentB_param_1.0.abc}[0]` allowed
        `${componentB_param_1.0.abc.0}` allowed

- nested parameters must be recursively resolved from innermost to outermost expression.

**Comment by namsonx** [comment]
Hello Thomas,

This enhancement was implemented in the [commit 45f79386](https://github.com/test-fullautomation/python-jsonpreprocessor/commit/45f793867fed297705633daef5ab09c809792c7b) on stabi branch ([namsonx/task/stabi_branch](https://github.com/test-fullautomation/python-jsonpreprocessor/tree/namsonx/task/stabi_branch))

Thank you,
Son

**Comment by test-fullautomation** [comment]
solved and already merged.
